### DISCUSSION TITLE
Azure Cache for Redis Premium in cluster mode

### DISCUSSION THREAD

**Discussion by maonat** [redisson/redisson#6234] [Q&A]
Hi everyone,

my team has Redisson open source implemented to connect to an Azure Cache for Redis in Premium tier with multiple shards enabled.

The performances are kinda disappointing and end up overloading the CPU of the entire shards with a strange trend.

![azure_dashboard](https://github.com/user-attachments/assets/39374574-6e99-4c09-b216-e4c515f9bb2d)

I saw that there are two ways of connecting to redis with Redisson:

1. Using [Single mode](https://redisson.org/docs/configuration/#single-yaml-config-format)
2. Using [Cluster mode](https://redisson.org/docs/configuration/#cluster-yaml-config-format)

Knowing that Azure cache for redis will allow only one single endpoint and a single port to use for connecting to the service... Which would be the proper settings to se in the application? Should we use Single mode because we have one single server to configure and the load balancing is done in the back by Azure or should we set a single entry in the `nodeAddresses` setting and use the cluster mode?

Thanks!

**Comment by mrniko** [comment]
Did you get these graphs using a Cluster mode? CPU spike depends on operations/s and data amount.

**Reply by maonat** [reply]
Yes, this is using the cluster mode with a single entry as per written in [Cluster mode](https://redisson.org/docs/configuration/#cluster-yaml-config-format)

**Reply by mrniko** [reply]
you could just use an endpoint

**Reply by maonat** [reply]
This is to understand if we tuned wrongly the client and this is causing the very strange trend where one single shard is being used the most and the others follow but in reality the memory used is just 250MB per shard!

**Reply by mrniko** [reply]
It's because of uneven key distribution

**Reply by maonat** [reply]
I'll pass this info to engineering team and see what they says about this after some tests 👍
